src/test/java/com/example/damian/iracki/ticketBookingApp/ScreeningResourceTest.java

```diff
@@ -0,0 +1,36 @@
+package com.example.damian.iracki.ticketBookingApp;
+
+import com.example.damian.iracki.ticketBookingApp.entities.Movie;
+import com.example.damian.iracki.ticketBookingApp.entities.Screening;
+import com.example.damian.iracki.ticketBookingApp.entities.ScreeningRoom;
+import com.example.damian.iracki.ticketBookingApp.entities.Ticket;
+import com.example.damian.iracki.ticketBookingApp.enums.PaymentStatus;
+import com.example.damian.iracki.ticketBookingApp.enums.TypeOfTicket;
+import com.example.damian.iracki.ticketBookingApp.validators.TicketValidator;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class ScreeningResourceTest {
+
+    private final TicketValidator ticketValidator = new TicketValidator();
+
+    @Test
+    public void shouldThrowErrorWhenFirstLetterOfNameIsNotCapital() {
```

Comment:
Ten test określiłbyś jako test jednostkowy, czy integracyjny?
W jaki sposób najlepiej przetestować Springowy kontroler REST?
Co innego można / warto przetestować w tej aplikacji?

Reply:
Jest to raczej test jednostkowy bo sprawdza tylko jedną rzecz.
Moglibyśmy sprawdzić resztę walidacji lub czy miejsca odpowiednio ustawiają się na niedostępne.
Springowy kontroler można przetestować za pomocą biblioteki MockMvc. Pozwala uzyskać konkretny endpoint z poziomu kodu.